### PR TITLE
automate polkadot.js/api tests

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes

- Implement Mocha/Chai tests for testing polkadot.js/api functions.
- Created go test to start gossamer node and mocha tests.
- Added it-polkadotjs to Makefile for running these tests.

## Tests

Running go test:
```
go test ./tests/polkadotjs_test/...
```
OR: 
```
make it-polkadotjs
```
OR, you can run the interact with the javascript tests and run them manually:

These tests assume gossamer in running with websocket port 8546, so start a gossamer node:
```
make gossamer
./bin/gossamer init --force
./bin/gossamer   --key alice  --roles 4 --ws --wsport 8546 --rpc
``` 

These tests are NodeJS tests using Mocha/Chai framework for testing, so they can be run with `npm` or `yarn`. First install dependencies: 

```
cd tests/polkadotjs_test/
yarn
yarn install
```
Then run the tests:
```
yarn test
```
Review results and enjoy.

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- closes: #1228

### CODE REVIEW

File: tests/polkadotjs_test/test/test-polkadot.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
+const {describe} = require('mocha');
+const {expect} = require('chai');
+const { ApiPromise, WsProvider } = require('@polkadot/api');
+const sleep = (delay) => new Promise((resolve) => setTimeout(resolve, delay))
+const {Keyring } = require('@polkadot/keyring');
```
Comment (RyRy79261) [inline tests/polkadotjs_test/test/test-polkadot.js:5]: Please use standard `{ Prop }` spacing for these imports 

Additionally, please move Line 4 down, style guide wise all dependencies come first followed by declarations

**Review by RyRy79261 — CHANGES_REQUESTED** [review]
Code looks good, minor stylistic change.

The tests fail however, this is likely due to missing `node` dependencies, the command `npm || yarn` needs to be executed prior to running the mocha script.

The `.gitignore` will also need to be amended to ignore the `node_modules` folder

File: tests/polkadotjs_test/test/test-polkadot.js (hunk 1 above)
Comment (edwardmack) [reply tests/polkadotjs_test/test/test-polkadot.js]: Updated, thanks for guidance regarding javascript style, I imagine there is a lot I may be missing here.  I'm going to research and start using a linting utility, do you have any suggestions?  Seems like a lot are using ESLint.

**Comment by noot** [comment]
when I do `make it-polkadotjs` i get an error:
```
>  Running Integration Tests polkadot.js/api mode... 
HOSTNAME=0.0.0.0 MODE=polkadot go test ./tests/polkadotjs_test/... -timeout=5m -v
=== RUN   TestStartGossamerAndPolkadotAPI
    start_polkadotjs_test.go:36: starting gossamer for polkadot.js/api tests...
t=2021-02-09T18:37:39-0500 lvl=info msg="initializing gossamer..." pkg=test/utils cmd="/home/elizabeth/go/src/github.com/ChainSafe/gossamer/bin/gossamer init --config /home/elizabeth/go/src/github.com/ChainSafe/gossamer/tests/utils/config_babe_max_threshold.toml --basepath /tmp/TestStartGossamerAndPolkadotAPI/alice --genesis-raw /home/elizabeth/go/src/github.com/ChainSafe/gossamer/tests/utils/genesis_oneauth.json --force"
t=2021-02-09T18:37:43-0500 lvl=info msg="initialized gossamer!" pkg=test/utils node=0
t=2021-02-09T18:37:43-0500 lvl=info msg="starting gossamer..." pkg=test/utils cmd="/home/elizabeth/go/src/github.com/ChainSafe/gossamer/bin/gossamer --port 7000 --config /home/elizabeth/go/src/github.com/ChainSafe/gossamer/tests/utils/config_babe_max_threshold.toml --basepath /tmp/TestStartGossamerAndPolkadotAPI/alice --rpchost 0.0.0.0 --rpcport 8540 --rpcmods system,author,chain,state,dev --rpc --log info --roles 4 --key alice --ws --wsport 8546"
t=2021-02-09T18:37:48-0500 lvl=info msg="node started" pkg=test/utils key=alice cmd.Process.Pid=32188
    start_polkadotjs_test.go:51: 
        	Error Trace:	start_polkadotjs_test.go:51
        	Error:      	Received unexpected error:
        	            	exit status 1
        	Test:       	TestStartGossamerAndPolkadotAPI
--- FAIL: TestStartGossamerAndPolkadotAPI (15.20s)
FAIL
FAIL	github.com/ChainSafe/gossamer/tests/polkadotjs_test	15.264s
FAIL
Makefile:65: recipe for target 'it-polkadotjs' failed
make: *** [it-polkadotjs] Error 1
```

maybe there is something missing from the make command?

**Comment by edwardmack** [comment]
@noot re make polkadotjs, yes I need to add an install for yarn, and probably npm as well.  I'm not sure best way to do that, I assume a .sh script.  @RyRy79261 do you know how to do this?

re: test 1 failure, I've fixed that, I'm now reading chain from Name in genesis.json.  It seems that our configs can overwrite that.

re: test 2, this times out and fails if the node isn't generating new blocks.  I'm using a config with 
```
[core]
roles = 4
babe-authority = true
grandpa-authority = true
slot-duration = 500
babe-threshold-numerator = 1
babe-threshold-denominator = 1
```

**Comment by RyRy79261** [comment]
@edwardmack Sorry for the delay, regarding avoiding the need for dependencies, we could use `npx` in place of `npm/yarn` this pulls the dependencies down as need be for the command so no need for pre collected dependencies

**Comment by noot** [comment]
@edwardmack btw, could you add `make it-polkadotjs` to the CI?

**Comment by edwardmack** [comment]
I've updated this to use npx instead of yarn.  Also enabled the balances.transfer test since it's working now.

**Review by noot — APPROVED** [review]
nice work!

**Review by RyRy79261 — APPROVED** [review]
Epic!

**Pull request merged**